mitmproxy/net/http/headers.py

```diff
@@ -33,3 +34,39 @@ def assemble_content_type(type, subtype, parameters):
         return f"{type}/{subtype}"
     params = "; ".join(f"{k}={v}" for k, v in parameters.items())
     return f"{type}/{subtype}; {params}"
+
+
+def infer_content_encoding(content_type: str, content: bytes = b"") -> str:
```

Comment:
This is `_guess_encoding` with no behavioral changes and some minor readability improvements.